packages/next-swc/crates/next-core/js/src/entry/app/edge-page-bootstrap.ts

```diff
@@ -0,0 +1,103 @@
+import 'next/dist/esm/server/web/globals'
+import { adapter } from 'next/dist/server/web/adapter'
+import { RSC_VARY_HEADER } from 'next/dist/client/components/app-router-headers'
+import { IncrementalCache } from 'next/dist/server/lib/incremental-cache'
+import { renderToHTMLOrFlight } from 'next/dist/server/app-render/app-render'
+;('TURBOPACK { chunking-type: isolatedParallel }')
+import entry from 'APP_ENTRY'
+import BOOTSTRAP from 'APP_BOOTSTRAP'
+import { createManifests, installRequireAndChunkLoad } from './manifest'
+import type { NextRequest, NextFetchEvent } from 'next/server'
+import type { RenderOpts } from 'next/dist/server/app-render/types'
+import type { ParsedUrlQuery } from 'querystring'
+
+installRequireAndChunkLoad()
+
+Error.stackTraceLimit = 100
+
+const { clientReferenceManifest } = createManifests()
+
+const MIME_TEXT_HTML_UTF8 = 'text/html; charset=utf-8'
+
+async function render(request: NextRequest, event: NextFetchEvent) {
+  const renderOpt: Omit<
+    RenderOpts,
+    'App' | 'Document' | 'Component' | 'pathname'
+  > & { params: ParsedUrlQuery } = {
+    // TODO params
+    params: {},
+    supportsDynamicHTML: true,
+    dev: true,
+    buildId: 'development',
+    buildManifest: {
+      polyfillFiles: [],
+      rootMainFiles: BOOTSTRAP.filter((path) => path.endsWith('.js')),
+      devFiles: [],
+      ampDevFiles: [],
+      lowPriorityFiles: [],
+      pages: {
+        '/_app': [],
+      },
+      ampFirstPages: [],
+    },
+    ComponentMod: {
+      ...entry,
+      __next_app_require__: __next_require__,
+      pages: ['page.js'],
+    },
+    clientReferenceManifest,
+    runtime: 'nodejs',
+    serverComponents: true,
+    assetPrefix: '',
+    pageConfig: {},
+    reactLoadableManifest: {},
+    // TODO nextConfigOutput
+    nextConfigOutput: undefined,
+  }
+
+  const tranform = new TransformStream()
+  const response = new Response(tranform.readable)
+
+  let [, pathname, query] = /^([^?]*)(.*)$/.exec(request.url)!
+
+  const result = await renderToHTMLOrFlight(
+    // @ts-expect-error - TODO renderToHTMLOrFlight types should accept web platform types
+    request,
+    response,
+    pathname,
+    // TODO query
+    {},
+    renderOpt as any as RenderOpts
+  )
+
+  response.headers.append(
+    'Content-Type',
+    result.contentType() || MIME_TEXT_HTML_UTF8
+  )
+  response.headers.append('Vary', RSC_VARY_HEADER)
+
+  const writer = tranform.writable.getWriter()
+  result.pipe({
+    write: (chunk: Uint8Array) => writer.write(chunk),
+    end: () => writer.close(),
+    destroy: (reason?: Error) => writer.abort(reason),
+  })
+
+  return response
+}
+
+// adapter uses this to detect edge rendering
+self.__BUILD_MANIFEST = {}
```

Comment:
There's a `// TODO-APP: use explicit marker for this`. Could we at least change this to `self.__EDGE_RENDERING` so we don't break this in the future when updating the `isEdgeRendering` logic?

If it's too complicated, update both comments to refer to the same issue on linear.

Reply:
I have no idea where to add this in next.js. I keep as TODo